common/coin_support/coin_utils.c

```diff
@@ -596,4 +596,25 @@ uint16_t get_account_name(const uint32_t *path, uint16_t account_type, char *acc
     }
 
     return length;
-}
\ No newline at end of file
+}
+
+void derivation_path_array_to_string(const uint32_t *path,
+                                     const size_t path_length,
+                                     const bool harden_all,
+                                     char *output,
+                                     const size_t out_len) {
+    if (out_len == 0 || output == NULL)
+        return;
```

Comment:
there should be a way to indicate that the output is invalid. currently, this results into garbage output and there is now way for the caller to distinguish between a valid output and an invalid output.

options:
1. memzero the entire output buffer
2. return the length of the output string
3. memzero the entire output buffer + return the length of the output string

Reply:
error code have been added